src/components/Patient/ManagePatients.tsx

```diff
@@ -291,8 +291,34 @@ export const PatientManager = () => {
     return cleanedData;
   };
 
+  // solving the issue Unnecessary API Calls and Re-Renders on Patient Page #9308
+  const [paramValue, setParamValue] = useState(params);
+  const previousValueRef = useRef(params);
+
+  function compareObjects(obj1: object, obj2: object) {
```

Comment:
If you console 'params' you observe that two params coming in differently but looking the same were being treated as different values. This was causing unnecessary API calls. Upon examining the previous parameter value and the current incoming parameter value, I found they were different. This issue was occurring not only when typing inside the phone number field but also when changing the option, causing unnecessary API calls.

```
{page: 1, limit: 12, name: undefined, patient_no: undefined, is_active: 'True', …}
{page: '1', limit: 12, name: undefined, patient_no: undefined, is_active: 'True', …}
```
If you observe the page key value, the previous one is a number, and the current one is a string. This was the problem. I solved it by comparing two objects while ignoring their value types, which now prevents unnecessary API calls